### PR TITLE
feat: Implement ITK recipe

### PR DESCRIPTION
**Pull request by arcadien**
Specify library name and version:  **itk/4.13.2**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by arcadien** [comment]
@lasote i cannot download the build log, and hence cannot fix the recipe. I hold on capacity of log access

**Comment by arcadien** [comment]
> 
> 
> @lasote i cannot download the build log, and hence cannot fix the recipe. I hold on capacity of log access

Well it works in Chrome but not in Firefox.

File: recipes/itk/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def requirements(self):
+        self.requires("itk/{0}".format(self.version))
+		
+    def imports(self):
+        self.copy("*.dll", dst="bin", src="bin")
+        self.copy("*.dylib*", dst="bin", src="lib")
+        self.copy("*.so*", dst="bin", src="lib")
+
+    def test(self):
```
Comment (Croydon) [inline recipes/itk/all/test_package/conanfile.py]: Please simplify this to use the `run_environment` parameter and exclude the execution of this test_package in the case of cross compiling, like this:

https://github.com/bincrafters/templates/blob/d331c6389ab1e0f898db57d252aa22f15a5e108d/default/test_package/conanfile.py#L15

File: recipes/itk/all/patches/CMakeProjectWrapper.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required(VERSION 3.0)
+project(cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(WIN32)
```
Comment (Croydon) [inline recipes/itk/all/patches/CMakeProjectWrapper.txt]: This looks like an error of some kind to me. This should definitely not be necessary

File: recipes/itk/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                pack_names.append("libpng-dev")
+            installer = tools.SystemPackageTool()
+            for p in pack_names:
+                installer.install(p)
+
+    def system_requirements(self):
+        if tools.os_info.linux_distro == "linuxmint":
+            pack_names = [
+                "libexpat1",
+                "libjpeg-turbo8",
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]: We do have a `libjpeg-turbo` recipe. Use this instead

File: recipes/itk/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+            elif tools.os_info.os_version.major(fill=False) == "19":
+                pack_names.append("libpng-dev")
+            installer = tools.SystemPackageTool()
+            for p in pack_names:
+                installer.install(p)
+
+    def system_requirements(self):
+        if tools.os_info.linux_distro == "linuxmint":
+            pack_names = [
+                "libexpat1",
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]: We do have an `expat` recipe, please use this instead

File: recipes/itk/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+                "libjpeg-turbo8",
+                "libtiff5"
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]: We do have a `libtiff` recipe, please us this instead

File: recipes/itk/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+            ]
+            if tools.os_info.os_version.major(fill=False) == "18":
+                pack_names.append("libpng12-0")
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]: We do have a `libpng` recipe, please us this instead

File: recipes/itk/all/patches/CMakeProjectWrapper.txt (hunk 2 above)
Comment (arcadien) [reply recipes/itk/all/patches/CMakeProjectWrapper.txt]: done

File: recipes/itk/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+            ]
+            if tools.os_info.os_version.major(fill=False) == "18":
+                pack_names.append("libpng12-dev")
+            elif tools.os_info.os_version.major(fill=False) == "19":
+                pack_names.append("libpng-dev")
+            installer = tools.SystemPackageTool()
+            for p in pack_names:
+                installer.install(p)
+
+    def system_requirements(self):
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]: You should be able to just remove this entire block

File: recipes/itk/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+        if not tools.os_info.is_linux:
+            self.requires("libjpeg/9c@bincrafters/stable")
+            self.requires("libpng/1.6.37@bincrafters/stable")
+            self.requires("libtiff/4.0.9@bincrafters/stable")
+
+        if tools.os_info.is_windows:
+            self.requires("Expat/2.2.6@pix4d/stable")
+            self.requires("zlib/1.2.11@conan/stable")
+
+    def build_requirements(self):
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]: You should be able to remove this block

File: recipes/itk/all/conanfile.py (hunk 3 above)
Comment (arcadien) [reply recipes/itk/all/conanfile.py]: done

File: recipes/itk/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+    def configure(self):
+        del self.settings.compiler.libcxx
+        if 'CI' not in os.environ:
+            os.environ["CONAN_SYSREQUIRES_MODE"] = "verify"
+
+    def requirements(self):
+
+        if not tools.os_info.is_linux:
+            self.requires("libjpeg/9c@bincrafters/stable")
+            self.requires("libpng/1.6.37@bincrafters/stable")
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]:
```suggestion
            self.requires("libpng/1.6.37")
```

File: recipes/itk/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        if 'CI' not in os.environ:
+            os.environ["CONAN_SYSREQUIRES_MODE"] = "verify"
+
+    def requirements(self):
+
+        if not tools.os_info.is_linux:
+            self.requires("libjpeg/9c@bincrafters/stable")
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]:
```suggestion
            self.requires("libjpeg/9c")
```

File: recipes/itk/all/conanfile.py (hunk 4 above)
Comment (arcadien) [reply recipes/itk/all/conanfile.py]: done

File: recipes/itk/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+            self.requires("libpng/1.6.37@bincrafters/stable")
+            self.requires("libtiff/4.0.9@bincrafters/stable")
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]:
```suggestion
            self.requires("libtiff/4.0.9")
```

File: recipes/itk/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+
+        if tools.os_info.is_windows:
+            self.requires("Expat/2.2.6@pix4d/stable")
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]:
```suggestion
            self.requires("expat/2.2.7")
```

File: recipes/itk/all/conanfile.py [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+            self.requires("zlib/1.2.11@conan/stable")
```
Comment (Croydon) [inline recipes/itk/all/conanfile.py]:
```suggestion
            self.requires("zlib/1.2.11")
```

Comment (arcadien) [reply recipes/itk/all/conanfile.py]: done

File: recipes/itk/all/conanfile.py (hunk 12 above)
Comment (arcadien) [reply recipes/itk/all/conanfile.py]: done

**Comment by arcadien** [comment]
all clear

**Comment by arcadien** [comment]
@Croydon used all Bincrafters dependencies, cleared the useless block dealing with Linux distros, removed the CMake wrapper (the Bincrafters zlib package may fix the problem worked around here) and simplified the conanfile.py of test_package. I can't do faster ;)

File: recipes/itk/all/conanfile.py [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+                '',
+                file_data,
+                re.M
+            )
+
+            # Write the file out again
+            with open(file_path, 'w') as file:
+                file.write(file_data)
+
+    def cmake_fix_path(self, file_path, package_name):
```
Comment (Morwenn) [inline recipes/itk/all/conanfile.py]: You can probably use the experimental [`patch_config_paths()`](https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#patch-config-paths-experimental) to achieve the same thing.

File: recipes/itk/all/test_package/CMakeLists.txt [hunk 15]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.7)
+project(example)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+include_directories(${CONAN_INCLUDE_DIRS_ITK_SUB})
```
Comment (uilianries) [inline recipes/itk/all/test_package/CMakeLists.txt]: You don't need to include the header for directly, conan_basic_setup should be able to detect the correct folder

Comment (arcadien) [reply recipes/itk/all/test_package/CMakeLists.txt]: i do use the hook. Its in the guidelines

**Comment by lasote** [comment]
Use requirement expat 2.2.7

File: recipes/itk/all/test_package/CMakeLists.txt (hunk 15 above)
Comment (arcadien) [reply recipes/itk/all/test_package/CMakeLists.txt]: There is a `CONAN_INCLUDE_DIRS_ITK` which is automatically managed. But for now, using `CONAN_INCLUDE_DIRS_ITK_SUB` manually is mandatory. Without it, it is impossible to `#include` any of the VNL headers included in ITK.

File: recipes/itk/all/conanfile.py [hunk 16]
```diff
@@ -0,0 +1,241 @@
...
+        self.cpp_info.includedirs = ['include']
+        # This hack allow to include directly, for example, a vnl
+        # file using `#include <vln/...>`
+        # Without the hack, the vln files cannot themselves include
+        # files using `#include<vnl/..>` since only <ITK-xx/vnl/...> is
+        # available. More, one has to add in CMakeLists.txt
+        # include_directories(${CONAN_INCLUDE_DIRS_ITK_SUB})
+        # to finish the use of this hack.
+        self.cpp_info.sub.includedirs = ['include/ITK-{0}'.format(
+            self.upstream_version)]
```
Comment (arcadien) [inline recipes/itk/all/conanfile.py]: @uilianries This is the other side of `include_directories(${CONAN_INCLUDE_DIRS_ITK_SUB})` in the test package. I agree this is not clever, but until now i have no other solution to let VNL components to be `#include`-ed.

Comment (uilianries) [reply recipes/itk/all/conanfile.py]: @arcadien thanks for clarifying

File: recipes/itk/all/conanfile.py [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+
+class LibITKConan(ConanFile):
+    name = "itk"
+    upstream_version = "4.13"
+    upstream_patch = "2"
+    version = "{0}.{1}".format(upstream_version, upstream_patch)
+
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False]}
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
    options = {"shared": [True, False], "fPIC": [True, False]}
```

File: recipes/itk/all/conanfile.py [hunk 18]
```diff
@@ -0,0 +1,241 @@
...
+
+    default_options = {"shared": True}
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
    default_options = {"shared": True, "fPIC": True}
```

File: recipes/itk/all/conanfile.py [hunk 19]
```diff
@@ -0,0 +1,241 @@
...
+    homepage = "http://www.itk.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "http://www.itk.org/licensing/"
+    description = "Insight Segmentation and Registration Toolkit"
+    source_subfolder = "sf"
+    build_subfolder = "build"
+    short_paths = True
+
+    def configure(self):
+        del self.settings.compiler.libcxx
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
        del self.settings.compiler.libcxx
        del self.settings.compiler.cppstd
```

File: recipes/itk/all/conanfile.py [hunk 20]
```diff
@@ -0,0 +1,241 @@
...
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False]}
+
+    default_options = {"shared": True}
+
+    homepage = "http://www.itk.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "http://www.itk.org/licensing/"
+    description = "Insight Segmentation and Registration Toolkit"
+    source_subfolder = "sf"
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
    _source_subfolder = "sf"
```

File: recipes/itk/all/conanfile.py [hunk 21]
```diff
@@ -0,0 +1,241 @@
...
+    build_subfolder = "build"
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
    _build_subfolder = "build"
```

File: recipes/itk/all/conanfile.py [hunk 22]
```diff
@@ -0,0 +1,241 @@
...
+        cmake.definitions["Module_ITKDICOMParser"] = "OFF"
+
+        # Disabled on Linux (link errors)
+        cmake.definitions["Module_ITKLevelSetsv4Visualization"] = "OFF"
+        cmake.definitions["Module_ITKVtkGlue"] = "OFF"
+
+        # Disabled because Vxl vidl is not build anymore
+        cmake.definitions["Module_ITKVideoBridgeVXL"] = "OFF"
+
+        if not tools.os_info.is_windows:
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
```

File: recipes/itk/all/conanfile.py [hunk 23]
```diff
@@ -0,0 +1,241 @@
...
+            cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = "ON"
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
```
fPIC option manages it automatically

File: recipes/itk/all/conanfile.py [hunk 24]
```diff
@@ -0,0 +1,241 @@
...
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("ITK-{0}.{1}".format(
+            self.upstream_version,
+            self.upstream_patch),
+            self.source_subfolder)
+
+    def build(self):
+        itk_source_dir = os.path.join(
+            self.source_folder,
+            self.source_subfolder)
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
            self._source_subfolder)
```

File: recipes/itk/all/conanfile.py [hunk 25]
```diff
@@ -0,0 +1,241 @@
...
+        self.requires("libpng/1.6.37")
+        self.requires("libjpeg/9c")
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("ITK-{0}.{1}".format(
+            self.upstream_version,
+            self.upstream_patch),
+            self.source_subfolder)
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
            self._source_subfolder)
```

File: recipes/itk/all/conanfile.py [hunk 26]
```diff
@@ -0,0 +1,241 @@
...
+        # Disabled on Linux (link errors)
+        cmake.definitions["Module_ITKLevelSetsv4Visualization"] = "OFF"
+        cmake.definitions["Module_ITKVtkGlue"] = "OFF"
+
+        # Disabled because Vxl vidl is not build anymore
+        cmake.definitions["Module_ITKVideoBridgeVXL"] = "OFF"
+
+        if not tools.os_info.is_windows:
+            cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = "ON"
+        cmake.configure(build_folder=self.build_subfolder)
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
        cmake.configure(build_folder=self._build_subfolder)
```

File: recipes/itk/all/conanfile.py [hunk 27]
```diff
@@ -0,0 +1,241 @@
...
+        cmake.build()
+        cmake.install()
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
```
It should be moved to the package stage

File: recipes/itk/all/conanfile.py [hunk 28]
```diff
@@ -0,0 +1,241 @@
...
+            tools.replace_in_file(
+                file_path,
+                self.deps_cpp_info[package_name].rootpath.replace('\\', '/'),
+                "${CONAN_" + package_name.upper() + "_ROOT}",
+                strict=False
+            )
+        except:
+            self.output.info("Ignoring {0}...".format(package_name))
+
+    def package(self):
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
    def package(self):
        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
        cmake = CMake(self)
        cmake.install()
```

File: recipes/itk/all/conanfile.py [hunk 29]
```diff
@@ -0,0 +1,241 @@
...
+        for path, subdirs, names in os.walk(self.package_folder):
+            for name in names:
+                if fnmatch(name, '*.cmake'):
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]: Cmake files are not allowed in package folder https://github.com/conan-io/conan-center-index/wiki/FAQ#why-are-cmake-findconfig-files-and-pkg-config-files-not-packaged

File: recipes/itk/all/test_package/conanfile.py [hunk 30]
```diff
@@ -0,0 +1,27 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class HelloTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    name = "example"
```
Comment (uilianries) [inline recipes/itk/all/test_package/conanfile.py]:
```suggestion
```

File: recipes/itk/all/test_package/conanfile.py [hunk 31]
```diff
@@ -0,0 +1,27 @@
...
+    version = "4.13.2"
```
Comment (uilianries) [inline recipes/itk/all/test_package/conanfile.py]:
```suggestion   
```

File: recipes/itk/all/test_package/conanfile.py [hunk 32]
```diff
@@ -0,0 +1,27 @@
...
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def requirements(self):
+        self.requires("itk/{0}".format(self.version))
+
+    def imports(self):
```
Comment (uilianries) [inline recipes/itk/all/test_package/conanfile.py]:
```suggestion
```

File: recipes/itk/all/test_package/conanfile.py [hunk 33]
```diff
@@ -0,0 +1,27 @@
...
+        self.copy("*.dll", dst="bin", src="bin")
```
Comment (uilianries) [inline recipes/itk/all/test_package/conanfile.py]:
```suggestion
```

File: recipes/itk/all/test_package/conanfile.py [hunk 34]
```diff
@@ -0,0 +1,27 @@
...
+        self.copy("*.dylib*", dst="bin", src="lib")
```
Comment (uilianries) [inline recipes/itk/all/test_package/conanfile.py]:
```suggestion
```

File: recipes/itk/all/test_package/conanfile.py [hunk 35]
```diff
@@ -0,0 +1,27 @@
...
+        self.copy("*.so*", dst="bin", src="lib")
```
Comment (uilianries) [inline recipes/itk/all/test_package/conanfile.py]:
```suggestion
```

**Comment by lasote** [comment]
checking

**Comment by lasote** [comment]
```
  File "/home/conan/workspace/conan-center-pull-request_PR-194/recipes/itk/all/conanfile.py", line 31
    del self.settings.compiler.cppstd
                                    ^
TabError: inconsistent use of tabs and spaces in indentation
```

**Comment by arcadien** [comment]
@lasote hello the build failed because Expat is not found, but the package is in the dependencies. It there some extra step to implement to have it recognized?

**Comment by lasote** [comment]
We spotted a bug that is causing the system to generate/merge incomplete revisions. We would like to understand first why and then I can regenerate the expat package for you. Thanks for your patience.

File: recipes/itk/all/conanfile.py [hunk 36]
```diff
@@ -0,0 +1,217 @@
...
+
+    def build(self):
+        itk_source_dir = os.path.join(
+            self.source_folder,
+            self._source_subfolder)
+
+        cmake = CMake(self)
+
+        cmake.definitions["BUILD_EXAMPLES"] = "OFF"
+        cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```
Comment (Morwenn) [inline recipes/itk/all/conanfile.py]: That one should be automatically set by the CMake build helper, you don't need to set it by hand.

**Comment by Morwenn** [comment]
I use my own recipe for ITK at work so I checked yours and we seem to have a few differences, some because my recipe is for ITK 5 but maybe not all of them.

Most notably I am using three more dependencies available directly from conan-center: eigen, fftw and double-conversion. Eigen is specific to ITK 5 but I believe that the other ones were usable with ITK 4 - correct me if I'm wrong.

**Comment by arcadien** [comment]
> 
> 
> I use my own recipe for ITK at work so I checked yours and we seem to have a few differences, some because my recipe is for ITK 5 but maybe not all of them.
> 
> Most notably I am using three more dependencies available directly from conan-center: eigen, fftw and double-conversion. Eigen is specific to ITK 5 but I believe that the other ones were usable with ITK 4 - correct me if I'm wrong.

FFTW is present in the `CMakeLists.txt`, so, yes it can be used. That's more a "political" choice: here we are also using the OpenCV/ITK glue code. I could add them as package options.
- OpenCV (We are using 3.4.4, but more recent should work)
- FFTW
- Intel's MKL can also be a good option
- I missed double-conversion in the CMakeLists, is the option deeper in the CMake scripts?
- Eigen does not appear as option in `CMakeLists.txt` for 4.13. For now this recipe if for 4.13
> 
> Also does ITK really use libjpeg-turbo? I couldn't find any reference to it in its build files.

AFAIK libjpeg-turbo is a drop-in replacement for libjpeg, with enhancement for advanced architectures. So there should be no reference, because it implements the libjpeg API.

Thanks for your comments i will update the PR this way.

**Comment by Morwenn** [comment]
Oh, I didn't know that libjpeg-turbo was a drop-in replacement. Since several libraries seemed to have dedicated support in their build options I thought that it had a different API, thanks for the explanation!

File: recipes/itk/all/conanfile.py (hunk 36 above)
Comment (arcadien) [reply recipes/itk/all/conanfile.py]: done

**Comment by arcadien** [comment]
I will hold until dependency package (eg Expat, Jpeg) become available again

**Comment by lasote** [comment]
expat fixed, relaunching to see how it goes.

**Comment by arcadien** [comment]
well @lasote I now have a problem with undetected libjpeg.

**Comment by lasote** [comment]
But the package of libjpeg is downloaded correctly, this is an issue with the cmake find package/name or something probably? @uilianries can you assist?

**Comment by Morwenn** [comment]
Also note that the found zlib is the system one and not the Conan one.

File: recipes/itk/all/conanfile.py [hunk 37]
```diff
@@ -0,0 +1,250 @@
...
+        "shared": [True, False],
+        "use_jpegturbo": [True, False],
+        "use_vtk": [True, False],
+        "use_fftw_float": [True, False],
+        "use_fftw_double": [True, False],
+        "use_opencv": [True, False],
+        "use_mkl": [True, False]}
+
+    default_options = {
+        "shared": True,
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
        "shared": True,
        "fPIC": True,
```

File: recipes/itk/all/test_package/conanfile.py [hunk 38]
```diff
@@ -0,0 +1,26 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class HelloTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    version = "4.13.2"
```
Comment (uilianries) [inline recipes/itk/all/test_package/conanfile.py]:
```suggestion
```
name and version are not required for test_package

File: recipes/itk/all/test_package/conanfile.py (hunk 32 above)
Comment (uilianries) [reply recipes/itk/all/test_package/conanfile.py]: Is there any reason for keeping imports? If not, remove it. The `run_environment=True` is able to solve all dynamic libs paths.

File: recipes/itk/all/conanfile.py [hunk 39]
```diff
@@ -0,0 +1,250 @@
+import os
+import re
+
+from conans import ConanFile, CMake, tools
+
+class LibITKConan(ConanFile):
+    name = "itk"
+
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/itk/all/conanfile.py]:
```suggestion
    generators = "cmake", "cmake_find_package"
```
This will solve your problem related to zlib

**Comment by arcadien** [comment]
Please follow-up in https://github.com/conan-io/conan-center-index/pull/262, I leaved my company today and my write access on this repository will be soon removed.

**Comment by arcadien** [comment]
Please follow-up in https://github.com/conan-io/conan-center-index/pull/262, I leaved my company today and my write access on this repository will be soon removed.

**Pull request closed without merging**